### ISSUE TITLE
Docs unclear: how to handle errors / what is "the `error` instance"?

### ISSUE THREAD

**Issue by ericdand** [octokit/request.js#347]
Hi there,

I'm trying to use the octokit library but I'm unable to find any documentation of what happens when an API error occurs, or how to handle such an error.

In my particular case, I'm trying to create a branch (a "git ref", in the parlance) which sometimes already exists. When calling `octokit.git.createRef` for a ref which already exists, I get an unhelpful error message buried in the raw Actions log ("[error]Reference already exists"), and my code is unceremoniously terminated.

I've tried wrapping the call in a try-catch to no avail:

```
try {
    resp = await octokit.git.createRef(
        owner: owner,
        repo: repo,
        ref: 'ref/heads/' + branchName,
        sha: currentSha
    );
    console.log(resp.status);
} catch (e) {
    console.log(e.status);
}
```

The [documentation in the README](https://github.com/octokit/request.js#request) for this repo states the following: 

> If an error occurs, the `error` instance has additional properties to help with debugging
> 
> - error.status The http response status code
> - error.headers The http response headers as an object
> - error.request The request options such as method, url and data
>

I had assumed that "the `error` instance" meant the error thrown by the promise returned by `createRef`, but it appears to not be so. 

**How should library users handle API errors?**

**Comment by gr2m** [comment]
> I had assumed that "the `error` instance" meant the error thrown by the promise returned by `createRef`, but it appears to not be so.

that's what it means. You can access `error.request` to see the request options that resulted in the error. 

> When calling `octokit.git.createRef` for a ref which already exists, I get an unhelpful error message buried in the raw Actions log ("[error]Reference already exists"), and my code is unceremoniously terminated.

I think the error message is pretty clear in this case? There is not more information we get from GitHub's API response

**Comment by ericdand** [comment]
Gregor, thanks for taking a look! Appreciate the attention.

> > I had assumed that "the `error` instance" meant the error thrown by the promise returned by `createRef`, but it appears to not be so.
> 
> that's what it means.

Interesting -- thanks for confirming that that is indeed what's supposed to happen. My code didn't seem to catch it properly, but perhaps I need to go back and make sure that GitHub Actions was truly using the latest version of my code, where I added the catch... I'll report back. It's probably just my own error.

> I think the error message is pretty clear in this case? There is not more information we get from GitHub's API response

Yes, the error is clear in itself, though very terse. I would have had no idea what it meant if my code hadn't had extensive debug logging itself to provide additional context. I would have expected to see the name of the API call which resulted in the error... but anyway, I opened this issue about those thrown errors, not to offer unsolicited feedback on error message formatting. ;)
